stacker/util.py

```diff
@@ -448,3 +452,168 @@ def ensure_s3_bucket(s3_client, bucket_name, bucket_region):
             logger.exception("Error creating bucket %s. Error %s",
                              bucket_name, e.response)
             raise
+
+
+class SourceProcessor():
+    """Makes remote python package sources available in the running python
+       environment."""
+
+    def __init__(self, stacker_cache_dir=None):
+        """
+        Processes a config's list of package sources
+
+        Args:
+            stacker_cache_dir (string): Directory of stacker local cache.
+                                        Default to $HOME/.stacker
+        """
+        stacker_cache_dir = (stacker_cache_dir or
+                             os.path.join(os.environ['HOME'], '.stacker'))
+        package_cache_dir = os.path.join(stacker_cache_dir, 'packages')
+        self.stacker_cache_dir = stacker_cache_dir
+        self.package_cache_dir = package_cache_dir
+        self.create_cache_directories()
+
+    def create_cache_directories(self):
+        """Ensures that SourceProcessor cache directories exist.
+
+           Returns True to indicate the cache directories are present."""
+        if not os.path.isdir(self.package_cache_dir):
```

Comment:
Using `os.isdir` will only check that the path is a directory - if it's a file, or link, it'll still try and create the directory (and fail, so maybe this is fine) - just wanted to make sure you've thought of this.